be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -675,12 +674,14 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
     // If _overflow_read_chunk contains some rows, it means that in previous round of `_do_get_next`,
     // _read_chunk reads more rows than its capacity, the overflow rows in saved in _overflow_read_chunk.
     // Here we reload those overflow rows to current context's _read_chunk.
-    if (_overflow_read_chunk != nullptr && !_overflow_read_chunk->is_empty()) {
-        _context->_read_chunk->swap_chunk(*_overflow_read_chunk);
+    if (_context->_overflow_read_chunk != nullptr && !_context->_overflow_read_chunk->is_empty()) {
```

Comment:
```
    // remove (logical) deleted rows.
    if (chunk_size > 0 && chunk->delete_state() != DEL_NOT_SATISFIED && !_opts.delete_predicates.empty()) {
```
does _overflow_read_chunk need handle delete state?

Reply:
i think it needs to

Reply:
current code haven't dealt with this yet